src/AbstractDateDropdown.php

```diff
@@ -18,34 +19,29 @@
  *     null_on_all_empty?: bool,
  *     ...
  * }
+ * @psalm-type InputArray = array<string, string>
  * @template TOptions of Options
  * @template-extends AbstractFilter<TOptions>
+ * @template TInput of array
  */
 abstract class AbstractDateDropdown extends AbstractFilter
 {
     /**
      * If true, the filter will return null if any date field is empty
-     *
-     * @var bool
      */
-    protected $nullOnEmpty = false;
+    protected bool $nullOnEmpty = false;
```

Comment:
Not sure if we should add native types to properties: I'm personally OK with it, but it leads to BC breaks downstream, if anybody did override this.

Reply:
Ah, never mind, this is for the next major.

At that point, I'd just get rid of `protected`, but that's for another patch.